src/pb_memory.cc

```diff
@@ -141,15 +141,36 @@ PbMemory::CopyBuffer(
     kind = cudaMemcpyDeviceToDevice;
   }
 
-  cudaError_t err =
-      cudaMemcpy(dst->DataPtr(), src->DataPtr(), src->ByteSize(), kind);
+  cudaError_t err;
+  if ((src->MemoryType() == TRITONSERVER_MEMORY_GPU &&
+       dst->MemoryType() == TRITONSERVER_MEMORY_GPU) &&
+      (src->MemoryTypeId() != dst->MemoryTypeId())) {
+    err = cudaMemcpyPeer(
```

Comment:
Is this if condition necessary? Do you have a test that triggers this if condition?

Reply:
Yes, to both questions. `cudaMemcpyPeer` needs to used when UVA is not enabled between different GPUs.

https://stackoverflow.com/questions/22694518/what-is-the-difference-between-cudamemcpy-and-cudamemcpypeer-for-p2p-copy

I think [L0_io](https://github.com/triton-inference-server/server/tree/main/qa/L0_io) test would trigger this condition. I'll double check to confirm.